File: 0121-best-time-to-buy-and-sell-stock/solution.java
class Solution {
    public int maxProfit(int[] prices) {
        int p=0,min=0;
        for(int i=0;i<prices.length;i++){
            if(prices[min]>prices[i])
            min=i; 
            if(prices[i]-prices[min]>p)
            p=prices[i]-prices[min];
        }
        return p;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        least=prices[0]
        max_pr=0
        for i in range(1,len(prices)):
            cur_pr=prices[i]-least
            max_pr=max(cur_pr,max_pr)
            least=min(prices[i],least)
        return max_pr
